src/common/dump.cpp

```diff
@@ -44,6 +44,44 @@ void dump_to_xflash(void) {
     w25x_fetch_error();
 }
 
+void dump_err_to_xflash(const char *error, const char *title) {
+    dumpmessage_t dump_message;
+    w25x_sector_erase(w25x_error_start_adress);
+    dump_message.flag = 0;
+    strlcpy(dump_message.msg_title, title, sizeof(dump_message.msg_title));
+    strlcpy(dump_message.msg_content, error, sizeof(dump_message.msg_content));
+    w25x_program(w25x_error_start_adress, (uint8_t *)(&dump_message), sizeof(dumpmessage_t));
+    w25x_fetch_error();
+}
+
+int dump_in_xflash_get_message(char *msg_dst, char *title_dst) {
+    dumpmessage_t dump_message;
```

Comment:
I would suggest this function to take `dumpmessage_t dump_message` by reference instead of taking pointers to char arrays. That way you avoid passing too small array by `char *msg_dst` or `char *title_dst` which may overflow, as the size is not passed nor checked.

Alternatively if you don't want to expose message flags, you can pass reference to fixed size arrays `char (&msg_dst)[DUMP_MSG_MAX_LEN]` and `char title_dst [DUMP_MSG_TITLE_MAX_LEN]` https://www.nextptr.com/question/a6212599/passing-cplusplus-arrays-to-function-by-reference

Alternatively you can use c-style and pass length of each array after each `char *msg_dst` and `char *title_dst` parameter.

Reply:
This feels utterly unnecessary, since strlcpy copies up to `sizeof(c-string ptr, not decayed to char ptr)` characters and it ensures, that it will be NULL-terminated.

There is no need of passed string size, because only size of buffer is important here. We want developers to write whatever msg they want. If it is longer - it will be truncated - if it is shorter (most of the time) - it will fill up only part of the buffered space and strlcpy ensures that.

Reply:
Yes, sizes are passed in `7c21391`.